src/components/collection-card/components/company-logo/index.tsx

```diff
@@ -14,26 +15,58 @@ export default function CompanyLogo({ name }: CompanyLogoProps): JSX.Element {
 		return <></>
 	}
 
-	const src =
-		name === 'terraform-cloud'
-			? require('@hashicorp/mktg-logos/product/terraform-cloud/primary/color.svg?url')
-			: require(`./img/${name}.png?url`)
+	const src = {
+		light:
+			name === 'terraform-cloud'
+				? require('@hashicorp/mktg-logos/product/terraform-cloud/primary/color.svg?url')
+				: require(`./img/${name}.png?url`),
+		dark:
+			name === 'terraform-cloud'
+				? require('@hashicorp/mktg-logos/product/terraform-cloud/primary/colorwhite.svg?url')
+				: require(`./img/${name}_dm.png?url`),
+	}
 
 	return (
 		<span className={s.root}>
-			<span className={classNames(s.inner, s[`logo-${name}`])}>
-				<Image
-					src={src}
-					alt=""
-					/**
-					 * Note: for use in CollectionCard, consumer should provide meaningful
-					 * text using the card heading element.
-					 */
-					objectPosition="left center"
-					layout="fill"
-					objectFit="contain"
-				/>
-			</span>
+			<ThemedLogo
+				hideOnTheme={GlobalThemeOption.light}
+				src={src.dark}
+				name={name}
+			/>
+			<ThemedLogo
+				hideOnTheme={GlobalThemeOption.dark}
+				src={src.light}
+				name={name}
+			/>
+		</span>
+	)
+}
+
+function ThemedLogo({
```

Comment:
Could add a note here as a breadcrumb. 

Eventually we will lift up a global `ThemedImage` component. Took the skateboard approach here and duplicated the approach from https://github.com/hashicorp/dev-portal/pull/1752 for this specific context.